sklearn/gaussian_process/tests/test_kernels.py

```diff
@@ -71,8 +76,9 @@ def eval_kernel_for_theta(theta):
 def test_kernel_theta():
     """ Check that parameter vector theta of kernel is set correctly. """
     for kernel in kernels:
-        if isinstance(kernel, KernelOperator) \
-           or isinstance(kernel, Exponentiation):  # skip non-basic kernels
+        # skip non-basic kernels
+        if isinstance(kernel, (KernelOperator,  Exponentiation,
+                               SelectDimensionKernel)):
```

Comment:
The test retrieves parameters by reflection, but I encode the underlying parameters of the kernel differently since we have a kernel as well as a activedims.